src/client/game.cpp

```diff
@@ -3119,55 +3123,57 @@ void Game::processPlayerInteraction(f32 dtime, bool show_hud, bool show_debug)
 			client->setCrack(-1, v3s16(0, 0, 0));
 			runData.dig_time = 0.0;
 		}
-	} else if (runData.dig_instantly && input->getLeftReleased()) {
-		// Remove e.g. torches faster when clicking instead of holding LMB
+	} else if (runData.dig_instantly && wasKeyReleased(KeyType::DIG)) {
+		// Remove e.g. torches faster when clicking instead of holding dig button
 		runData.nodig_delay_timer = 0;
 		runData.dig_instantly = false;
 	}
 
-	if (!runData.digging && runData.ldown_for_dig && !input->getLeftState()) {
-		runData.ldown_for_dig = false;
-	}
+	if (!runData.digging && runData.btn_down_for_dig && !isKeyDown(KeyType::DIG))
+		runData.btn_down_for_dig = false;
```

Comment:
Not working properly either. Released the dig key in between and the crack does not disappear and the progress is kept.

![grafik](https://user-images.githubusercontent.com/1497498/84567801-54973600-ad7b-11ea-92f5-0f3244416a2f.png)

Reply:
Not sure if this is related, but I've come across this in master occasionally, especially when I'm trying to dig nodes right at the max range of the tool. I'll look into this, though. Thanks for reporting.

Reply:
Good catch, again. This can only be reproduced when binding dig to a normal key. Investigating...

Reply:
Fixed.